src/python/pants/engine/fs.py

```diff
@@ -300,7 +301,7 @@ def rules():
         QueryRule(Digest, (PathGlobs,)),
         QueryRule(Digest, (AddPrefix,)),
         QueryRule(Digest, (RemovePrefix,)),
-        QueryRule(Digest, (DownloadFile,)),
+        QueryRule(Digest, (DownloadFile,CompleteEnvironmentVars)),
```

Comment:
This does not excite me. Discussion in Slack: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1672279803640719